src/models/mod.rs

```diff
@@ -115,3 +117,71 @@ pub(crate) fn create_placeholder(
     item += ")";
     Ok(item)
 }
+
+pub async fn select_retry_or_panic(
+    pool: &sqlx::Pool<sqlx::Postgres>,
+    query: &str,
+    substitution_items: &[String],
+    retry_count: usize,
+) -> anyhow::Result<Vec<sqlx::postgres::PgRow>> {
+    let mut interval = crate::INTERVAL;
+    let mut retry_attempt = 0usize;
+
+    loop {
+        if retry_attempt == retry_count {
+            return Err(anyhow::anyhow!(
+                "Failed to perform query to database after {} attempts. Stop trying.",
+                retry_count
+            ));
+        }
+        retry_attempt += 1;
+
+        let mut args = sqlx::postgres::PgArguments::default();
+        for item in substitution_items {
+            args.add(item);
+        }
+
+        match sqlx::query_with(query, args).fetch_all(pool).await {
+            Ok(res) => return Ok(res),
+            Err(async_error) => {
+                // todo we print here select with non-filled placeholders. It would be better to get the final select statement here
+                tracing::error!(
+                         target: crate::LOGGING_PREFIX,
+                         "Error occurred during {}:\nFailed SELECT:\n{}\n Retrying in {} milliseconds...",
+                         async_error,
+                    query,
+                         interval.as_millis(),
+                     );
+                tokio::time::sleep(interval).await;
+                if interval < crate::MAX_DELAY_TIME {
+                    interval *= 2;
+                }
+            }
+        }
+    }
+}
+
+pub(crate) async fn start_after_interruption(
+    pool: &sqlx::Pool<sqlx::Postgres>,
+) -> anyhow::Result<u64> {
+    let query = "SELECT MAX(block_height)
```

Comment:
Union the NFT_EVENTS and COIN_EVENTS tables together and finds the max block height from those two.

Reply:
Why max block height?

Reply:
We should actually take MIN here. 

If in a given block height, the indexer has to write to both nft_events and coin_events and the indexer halts between the independent writes to the database. We should choose the minimum of the two block heights to catch this edge case although we end up indexing from the previous 1000 blocks, it is better to be correct. Thanks :) Just updated to take the MIN.